hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -919,24 +919,37 @@ class EmitMethodBuilder[C](
     }
   }
 
+  private def _invoke[T](_args: Param*): Code[T] = {
+    EmitCodeBuilder.scopedCode(this) { cb =>
+      val args = _args.map {
+        case cp: CodeParam =>
+          cp
+        case EmitParam(EmitCode(setup, m, pv)) =>
+          cb += setup
+          if (pv.pt.required) {
+            cb += Code.toUnit(m) // side effects???, is this even possible
+            EmitParam(EmitCode.present(pv))
+          } else {
+            val mv = cb.newField[Boolean](s"emitparam_m_${pv.pt._asIdent}", m)
+            EmitParam(EmitCode(Code._empty, mv, PCode(pv.pt, mv.mux(defaultValue(pv.pt), pv.code))))
+          }
+      }
+      mb.invoke(args.flatMap {
+        case CodeParam(c) => FastIndexedSeq(c)
+        case EmitParam(ec) => ec.codeTuple()
+      }: _*)
```

Comment:
Let's think about the code we want to generate here.  Imagine I have a code parameter p1 and EmitCode parameter p2, where p2 has code tuple types t1 ... tn.  I think what we want is:

```
local xm: Boolean = false
local x1: t1, ..., xn: tn
if p2.m
  xm = true
  (x1, ..., xn) = p2.codeTuple()
invoke(p1, xm, x1, ..., xn)
```

I don't see how you can do better than that.  So I'd flatMap over the (emit)Args to get the args for the non-emit invoke.